base/iterators.jl

```diff
@@ -1559,18 +1561,13 @@ Stacktrace:
     return ret
 end
 
-# Collections of known size
-only(x::Ref) = x[]
```

Comment:
While I recognize that you are trying to simplify the code, performing this simplification while you are trying to advance the other technical points makes the diff complicated. I think it might be good to focus on the additions first, and then make a second pull request for the simplification. This also would aide future debugging if there is an issue.